config/confighttp/confighttp.go

```diff
@@ -58,6 +58,19 @@ type HTTPClientSettings struct {
 
 	// Auth configuration for outgoing HTTP calls.
 	Auth *configauth.Authentication `mapstructure:"auth,omitempty"`
+
+	// MaxIdleConns is used to set a limit to the maximum idle HTTP connections the client can keep open.
+	MaxIdleConns *int `mapstructure:"max_idle_conns"`
```

Comment:
I understand the motivation to do this, but I think long term is better if we provide a "DefaultHTTPClientSettings" and users should use that in their default configs. Once we change all the contrib users we can make this a non-pointer.

Reply:
Could plz elaborate more as the current functionality is that we are cloning default transport configuration and updating transport parameters based on user inputs and configuring `httpclient` based on that.

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L52

Reply:
@bogdandrutu Could you plz give your inputs above?

Reply:
That is not public (users cannot use to initialize their configs). We want component's configs to initialize with that and we can start from a "clean" config, or not have to use pointers. Essentially I want to avoid having to use pointers in the future.

Reply:
Hello @bogdandrutu,

Okay, as mentioned by [you earlier](https://github.com/open-telemetry/opentelemetry-collector/pull/4188#discussion_r748475400), we will keep pointers as of now and also introduce a function `DefaultHTTPClientSettings`, which will be used by the component's config. Once all the contrib users use `DefaultHTTPClientSettings`, we can change those config options to non-pointers. I have tested this approach locally by making changes in the opentelemtry-collector repo and used those changes(used `DefaultHTTPClientSettings`) in the opentelemtry-collector-contrib repo and it works as expected.

Does it sound reasonable?

CC: @dmitryax, @rmfitzpatrick

Reply:
Can you please just open an issue (assign to you) to track the work left?

Reply:
I have opened one [issue](https://github.com/open-telemetry/opentelemetry-collector/issues/4513) and described the process in the description, also I am unable to assign it to myself.